File: backend/app/models/healthHabits.py
from typing import Literal

from pydantic import BaseModel


class HealthHabitsInput(BaseModel):
    sleep_hours: float
    sleep_quality: float
    caffeine_mg: float
    exercise_min: float
    mood_score: float
    social_interaction: float
    water_liters: float
    meal_regularity: float
    junk_food: float
    work_hours: float
    screen_time_h: float
    mindfulness: float


class StressPredictionOutput(BaseModel):
    predicted_stress_score: float
    stress_level: str


class BMIPlanInput(BaseModel):
    height_cm: float
    weight_kg: float
    goal: Literal["lose", "gain"]
    age: int | None = None
    sex: Literal["male", "female"] | None = None
    activity_level: Literal["sedentary", "light", "moderate", "active", "very_active"] | None = None
    diet: str | None = None
    exercise_focus: str | None = None
    daily_calorie_target: int | None = None


class BMIPlanOutput(BaseModel):
    bmi: float
    bmi_category: str
    goal: str
    target_weight_range_kg: str
    nutrition_plan: list[str]
    exercise_plan: list[str]
    recommendations: list[str]

File: health_mate_mobile_app/lib/models/health_habits.dart
class HealthHabitsInput {
  double sleepHours;
  int sleepQuality;
  double caffeineMg;
  double exerciseMin;
  double moodScore;
  int socialInteraction;
  double waterLiters;
  int mealRegularity;
  int junkFood;
  double workHours;
  double screenTimeH;
  int mindfulness;

  HealthHabitsInput({
    this.sleepHours = 7.0,
    this.sleepQuality = 5,
    this.caffeineMg = 100.0,
    this.exerciseMin = 30.0,
    this.moodScore = 6.0,
    this.socialInteraction = 5,
    this.waterLiters = 1.5,
    this.mealRegularity = 5,
    this.junkFood = 2,
    this.workHours = 8.0,
    this.screenTimeH = 4.0,
    this.mindfulness = 5,
  });

  Map<String, dynamic> toJson() => {
    'sleep_hours': sleepHours,
    'sleep_quality': sleepQuality,
    'caffeine_mg': caffeineMg,
    'exercise_min': exerciseMin,
    'mood_score': moodScore,
    'social_interaction': socialInteraction,
    'water_liters': waterLiters,
    'meal_regularity': mealRegularity,
    'junk_food': junkFood,
    'work_hours': workHours,
    'screen_time_h': screenTimeH,
    'mindfulness': mindfulness,
  };
}
